[docs] Update MSSQL README to use documentation guidelines (#8240)
elastic/integrations: packages/microsoft_sqlserver/_dev/build/docs/README.md

# Microsoft SQL Server Integration

The Microsoft SQL Server integration package allows you to search, observe, and visualize the SQL Server audit logs, as well as performance and transaction log metrics, through Elasticsearch.

> For more information on SQL Server auditing, refer to [SQL Server Audit](https://docs.microsoft.com/en-us/sql/relational-databases/security/auditing/sql-server-audit-database-engine?view=sql-server-ver15).

Reviewer: The line in the beginning paragraph, somewhat looks mis-placed, since it is referring the audit data-stream primarily. It may wrongly infer referring to all data streams.

## Data streams

The Microsoft SQL Server integration collects two types of data streams: logs and metrics.

**Logs** help you keep a record of events happening in Microsoft SQL Server.
Log data streams collected by the integration include:

* `audit` provides events from the configured Windows event log channel.
* `logs` parses error logs created by the Microsoft SQL server.

Other log sources, such as files, are not supported.

> Find more details in [Logs](#logs-reference).

Reviewer: Unable to click the #logs-reference, in the generated document. Can we verify the anchors exist?

<img width="706" alt="Screenshot 2023-10-31 at 11 23 41 AM" src="https://github.com/elastic/integrations/assets/69236064/04beb715-27b4-4594-ba9e-2e5a02a75b1f">

Reviewer 2: Fix:

s/logs-reference/logs

**Metrics** give you insight into the state of Microsoft SQL Server.
Metric data streams collected by the integration include:

* `performance` metrics gather the list of performance objects available on that server. Each server will have a different list of performance objects depending on the installed software.
* `transaction_log` metrics collect all usage stats and the total space usage.

> Find more details in [Metrics](#metrics-reference).

Reviewer 2: Fix:

s/metrics-reference/metrics

> * `performance` metrics gathers the list of performance objects available on that server. Each server will have a different list of performance objects depending on the installed software.
> * `transaction_log` metrics collects all usage stats and the total space usage.
> 
> Find more details in [Metrics](#metrics-reference).

Reviewer 2: ```suggestion
* `performance` metrics gather the list of performance objects available on that server. Each server will have a different list of performance objects depending on the installed software.
* `transaction_log` metrics collect all usage stats and the total space usage.

Find more details in [Metrics](#metrics-reference).
```

## Requirements

> You need Elasticsearch for storing and searching your data and Kibana for visualizing and managing it.
> You can use our hosted Elasticsearch Service on Elastic Cloud, which is recommended, or self-manage the Elastic Stack on your own hardware.

Reviewer 2: ```suggestion
You need Elasticsearch for storing and searching your data and Kibana for visualizing and managing it.
You can use our hosted Elasticsearch Service on Elastic Cloud, which is recommended, or self-manage the Elastic Stack on your hardware.
```

1. `transaction_log`:
    - [sys.databases](https://learn.microsoft.com/en-us/sql/relational-databases/system-compatibility-views/sys-sysdatabases-transact-sql?view=sql-server-ver16)
    - [sys.dm_db_log_space_usage](https://learn.microsoft.com/en-us/sql/relational-databases/system-dynamic-management-views/sys-dm-db-log-space-usage-transact-sql?view=sql-server-ver16)
    - [sys.dm_db_log_stats (DB_ID)](https://learn.microsoft.com/en-us/sql/relational-databases/system-dynamic-management-views/sys-dm-db-log-stats-transact-sql?view=sql-server-ver16)
2. `performance`:
    - [sys.dm_os_performance_counters](https://learn.microsoft.com/en-us/sql/relational-databases/system-dynamic-management-views/sys-dm-os-performance-counters-transact-sql?view=sql-server-ver16)

## Setup

For step-by-step instructions on how to set up any integration, refer to the
{{ url "getting-started-observability" "Getting started" }} guide.

Below you'll find more specific details on setting up the Microsoft SQL Server integration.

> <!-- Additional set up instructions -->

Reviewer: Is this formatting intended: it comes as below:

<img width="734" alt="Screenshot 2023-10-31 at 11 22 49 AM" src="https://github.com/elastic/integrations/assets/69236064/099eba8f-69b4-413d-a292-fe07f192d5a4">

Reviewer 2: https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#hiding-content-with-comments

Ideally, this should have worked. On GitHub it does. Maybe for the website, we are using a different markdown renderer that does not support this?

### Named Instance

Microsoft SQL Server has a feature that allows running multiple databases on the same host (or clustered hosts) with separate settings. Establish a named instance connection by using the instance name along with the hostname (e.g. `host/instance_name` or `host:named_instance_port`) to collect metrics. Details of the host configuration are provided below.

#### Query by Instance Name or Server Name in Kibana

The data can be visualized in Kibana by filtering based on the instance name and server name. The instance name can be filtered by `mssql.metrics.instance_name` and the server name by `mssql.metrics.server_name` fields.

### Host Configuration

As part of the input configuration, you need to provide the user name, password and host details. The host configuration supports both named instances or default (no-name) instances, using the syntax below.

> >Note: This integration supports collecting metrics from a single host. For multi-host metrics, each host can be run as a new integration.

Reviewer 2: ```suggestion
> Note: This integration supports collecting metrics from a single host. For multi-host metrics, each host can be run as a new integration.
```

**Connecting to Default Instance (host)**:

* `host`        (e.g. `localhost` (Instance name is not needed when connecting to default instance))
* `host:port`   (e.g. `localhost:1433`)

**Connecting to Named Instance (host)**:

* `host/instance_name`          (e.g. `localhost/namedinstance_01`)
* `host:named_instance_port`    (e.g. `localhost:60873`)

### Configuration

#### Audit

There are several levels of auditing for SQL Server, depending on government or standards requirements for your installation. The SQL Server Audit feature enables you to audit server-level and database-level groups of events and individual events. 

For more information on the different audit levels, refer to [SQL Server Audit Action Groups and Actions](https://docs.microsoft.com/en-us/sql/relational-databases/security/auditing/sql-server-audit-action-groups-and-actions?view=sql-server-ver15).
Then to enable auditing for SQL Server, refer to these [instructions](https://docs.microsoft.com/en-us/sql/relational-databases/security/auditing/create-a-server-audit-and-server-audit-specification?view=sql-server-ver15).

> >Note: For the integration package to be able to read and send audit events the event target must be configured to be Windows event log.

Reviewer 2: Isn't this gonna break the format? Isn't the syntax for blockquote is

```
> text
```

and not

```
>text
```

I'm not sure if `>text` works? Ref: https://www.markdownguide.org/basic-syntax/#blockquotes-1

Reviewer 2: I checked. As per [CommonMark](https://commonmark.org/) spec, the space is optional. So you can ignore this comment other similar ones.

Read more in [View the SQL Server error log in SQL Server Management Studio](https://docs.microsoft.com/en-us/sql/relational-databases/performance/view-the-sql-server-error-log-sql-server-management-studio?view=sql-server-ver16).

#### Performance metrics

Collects the `performance` counter metrics. The dynamic counter feature provides flexibility to collect metrics by providing the counter as an input.
This input can be a regular expression which will filter results based on pattern.
For example, if %grant% is given as input, it will enable metrics collection for all of the counters with names like 'Memory Grants Pending', 'Active memory grants count' etc.
MSSQL supports a limited set of regular expressions. For more details, refer to [Pattern Matching in Search Conditions](https://learn.microsoft.com/en-us/previous-versions/sql/sql-server-2008-r2/ms187489(v=sql.105)?redirectedfrom=MSDN).

> Note: Dynamic counters will go through some basic ingest pipeline post-processing to make counter names in lowercase and remove special characters and these fields will not have any static field mappings.

> The feature `merge_results` has been introduced in 8.4 beats which creates a single event by combining the metrics together in a single event. For more details, refer to [SQL module](https://www.elastic.co/guide/en/beats/metricbeat/current/metricbeat-module-sql.html#_example_merge_multiple_queries_to_single_event).
> 

Reviewer 2: ```suggestion
The feature `merge_results` has been introduced in 8.4 beats which creates a single event by combining the metrics in a single event. For more details, refer to [SQL module](https://www.elastic.co/guide/en/beats/metricbeat/current/metricbeat-module-sql.html#_example_merge_multiple_queries_to_single_event).

```

nit: together is redundant

> Read more in [Instructions about each performance counter metrics](https://docs.microsoft.com/en-us/sql/relational-databases/system-dynamic-management-views/sys-dm-os-performance-counters-transact-sql?view=sql-server-ver15).

Reviewer 2: ```suggestion
Read more in [instructions about each performance counter metrics](https://docs.microsoft.com/en-us/sql/relational-databases/system-dynamic-management-views/sys-dm-os-performance-counters-transact-sql?view=sql-server-ver15).
```

#### Transaction log metrics

Collects system level `transaction_log` metrics information for SQL Server instance.
Metrics for user-level databases can be collected by providing a list of user databases for which metrics are to be collected.

> Read more in [Instructions and the operations supported by transaction log](https://docs.microsoft.com/en-us/sql/relational-databases/logs/the-transaction-log-sql-server?view=sql-server-ver15).

Reviewer 2: ```suggestion
Read more in [instructions and the operations supported by transaction log](https://docs.microsoft.com/en-us/sql/relational-databases/logs/the-transaction-log-sql-server?view=sql-server-ver15).
```
